### DISCUSSION TITLE
Feedback from weekend training volunteers

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#316] [Ideas]
Everyone was very satisfied with BiblioteQ: ****easy to use and clear** **** their words: Congratulations to the author !**

The personalized query by keyboard shortcut is a marvel for people in difficulty!
Despite an underpowered processor (Athlon II - X2 2500) BiblioteQ displays data (> 4000) at superb speed.

We're working on a 1366x768 resolution laptop, so the resolution is a bit low for multiple windows, which causes a few problems.

Their comments and difficulties :

In Batch activities / Borrowing
BiblioteQ does not provide the borrower's name after entering the identifier.
BiblioteQ does not provide titles for books in response to barcode scans, but rather identifiers.

This gives them the impression of validating loans in a blur.

`Suggestion for improvement 1:`

In BA/Emrpunter have an editable pop-up which will be the list of "NAME - First name" with filtering to display only names beginning with the letters specify "D..." or "R...". (I have no idea how difficult this is, hence the need to open a discussion!)

or if number 1 is too complicated to do

`suggestion for improvement 2:`
In the member explorer, right-click on the item: "send member ID in the BA/Borrow member ID field".

this will save a lot of manipulation between the Member Manager / Batch Activities / Main Window screens and copy errors.

In Member Explorer
searches are made by member name, can the pop-up be reversed (NAME then Member ID to avoid having to do the same thing for each person)?

a French mistake is mine :-) correct in version of 2023-08-30.

**Comment by textbrowser** [comment]
"In Member Explorer
searches are made by member name, can the pop-up be reversed (NAME then Member ID to avoid having to do the same thing for each person)?"

What does that mean?

**Reply by meteos77** [reply]
Sorry for the lack of clarity :(
The pop-up shows Filter "Member ID" then in 2: "Last Name".

We do 99% of our searches on the person's "Last Name" and 1% on the "Member ID": "mxxxxx".

So if the pop-up were to show "Last Name" by default, we wouldn't have to click on the pop-up to tell it we were going to search by "Last Name".

**Reply by textbrowser** [reply]
That's one selection during a session.

**Reply by meteos77** [reply]
We have a small screen (portable) and to see the table, people close the members' window.

**Comment by meteos77** [comment]
If you could take a fresh look, that would be great.
import file model 1
[2023-08-27_énumérations-base-de-données-version1.csv](https://github.com/textbrowser/biblioteq/files/12455474/2023-08-27_enumerations-base-de-donnees-version1.csv)

**Comment by textbrowser** [comment]
It's broken and I can't spend days reviewing it. The original import was a 1-1 of the export. Now it has two other templates. Don't know. It's too complicated.

**Comment by textbrowser** [comment]
I really don't understand the need to have BQ do this when there are tools just for importing data. Every database revision requires a visit to this thing and a bunch of tests that break my interest.

**Reply by meteos77** [reply]
Because BiblioteQ is great!  Ok thanks, I'll see what else I can do

**Reply by textbrowser** [reply]
Completed.

**Reply by meteos77** [reply]
Super thanks, it works again.

**Comment by textbrowser** [comment]
"In Batch activities / Borrowing
BiblioteQ does not provide the borrower's name after entering the identifier.
BiblioteQ does not provide titles for books in response to barcode scans, but rather identifiers."

The item is in hand. This is one of the primary purposes of batch activities. The items are not on a shelf or in a make-believe land. We've discussed this to death. There is also Discover. So, I'm not about to duplicate that.

Member's name is lots of work. Cannot be added to the table because the table entries may be existing items or items which exist after an event. So, too late. The other place would be another query. BA is degrading into a duplicate item.

So, in order to reserve an item, in a normal universe: one has the items and one knows who the patron is.

**Reply by textbrowser** [reply]
I agree that if the member is unknown or vacant or dead or a ghost or in an alternate universe, the relationship between their name and their identifier would he purposeful. But they're there looking at you or sipping marijuana or doodling. Like, they know their identifier and if they don't, there is another panel to verify it. So, there's a lot of work when the solution is already presented.

**Reply by meteos77** [reply]
I'm well aware that it would be desirable to develop a card with a bar code for each user to avoid having to copy the identifier.

- A scan of the card and the direct member ID.

I've talked to them about it, so I hope we'll find a good solution.

**Reply by meteos77** [reply]
If you go to the supermarket, the checkout registers the item with its barcode, but displays the item with a clear name to the customer
2789878788974465 -> Lindt chocolate 70% ...

**Reply by textbrowser** [reply]
One would like everything present in a view that simplifies the nature of the action(s). Sure, but I only adapt that to new projects. BQ is aged like wine. It's fine. It doesn't have to adapt to all of the newness. The peacock approach is not necessary. In some cultures, perfection is unhappiness. Imperfection is more important.

**Reply by textbrowser** [reply]
![bq](https://github.com/textbrowser/biblioteq/assets/10701156/df150743-872c-4797-b3c9-d35af7ab1682)

**Reply by meteos77** [reply]
For me, Discover is the best!

**Comment by textbrowser** [comment]
> If you go to the supermarket, the checkout registers the item with its barcode, but displays the item with a clear name to the customer 2789878788974465 -> Lindt chocolate 70% ...



> If you go to the supermarket, the checkout registers the item with its barcode, but displays the item with a clear name to the customer 2789878788974465 -> Lindt chocolate 70% ...

Has been discussed. Items are in hand. Batch activities is not a data verifier. Also discussed.

**Comment by textbrowser** [comment]
If your approach is negative, BA is not suitable for you. If you approach it as it is, your outcome is happier.

**Comment by textbrowser** [comment]
Q. "Why doesn't it show me the item title?"
A. "You're holding the item."
Q. "But I have to read the code and make sure it's my item."
A. "You're trying to enter the wrong house with that key."
Q. "Software should help me be perfect."
A. "Some does, some  will not."

**Comment by meteos77** [comment]
Of course it is, 
Look at my first sentence

Everyone was very pleased with BiblioteQ: ease of use and clarity **** their words: Congratulations to the author!

Version 2023-06-06 is the one validated for our library.
Future developments are a plus to ease the transition, and I'm passing on their impressions to you, as I did with the "last name" popup, so that you can have some feedback.

**Reply by meteos77** [reply]
Don't forget that our library is made up of volunteers (> 65 years old), i.e. non-professionals, and they want to be reassured about what they're doing.

**Comment by textbrowser** [comment]
Will see if adding a title from a query spanning multiple tables is simple. If not, Discover. As for the patron's name, this has to exist separate of the table. New widgets. New queries. More work that's silly.

**Reply by meteos77** [reply]
You know that with my suggestions, I never know if it's a mountain ... now, "I think I understand the difficulty for the title.

**Comment by textbrowser** [comment]
Nice comments glossed by new requests turn me grumpy.

**Reply by meteos77** [reply]
It's real and sincere, forgot my request in this case.
